zmq4_pubsub_test.go

```diff
@@ -291,3 +291,63 @@ func TestPubSubClosedSub(t *testing.T) {
 		t.Fatalf("error: %+v", err)
 	}
 }
+
+func TestGetTopics(t *testing.T) {
+	var (
+		topics    = []string{"", "topic1", "topic2"}
+		getTopics []string
+	)
+
+	ctx, timeout := context.WithTimeout(context.Background(), 20*time.Second)
+	defer timeout()
+
+	ep := must(EndPoint("tcp"))
+	pub := zmq4.NewPub(ctx)
+	sub0 := zmq4.NewSub(ctx, zmq4.WithID(zmq4.SocketIdentity("sub0")))
+	sub1 := zmq4.NewSub(ctx, zmq4.WithID(zmq4.SocketIdentity("sub1")))
+	sub2 := zmq4.NewSub(ctx, zmq4.WithID(zmq4.SocketIdentity("sub2")))
+
+	subs := []zmq4.Socket{sub0, sub1, sub2}
+
+	defer pub.Close()
+	defer sub0.Close()
+	defer sub1.Close()
+	defer sub2.Close()
+
+	err := pub.Listen(ep)
+	if err != nil {
+		t.Errorf("could not listen: %+v", err)
+	}
+
+	for isub, sub := range subs {
+		err = sub.Dial(ep)
+		if err != nil {
+			t.Errorf("could not dial: %+v", err)
+		}
+
+		err = sub.SetOption(zmq4.OptionSubscribe, topics[isub])
+		if err != nil {
+			t.Errorf("could not subscribe to topic %q: %+v", topics[isub], err)
+		}
+		time.Sleep(500 * time.Millisecond)
+		getTopics = pub.(zmq4.Topics).Topics()
+		if len(getTopics) != isub+1 {
+			t.Errorf("got %d topics, want %d topics", len(getTopics), isub+1)
+		} else {
+			for _, wt := range topics[:isub+1] {
```

Comment:
instead of this, perhaps just compare against a sorted `[]string` ?
_e.g._
```go
want := make([]string, isub+1)
copy(want, topics)
sort.Strings(want)
sort.Strings(got)
if !reflect.DeepEqual(got, want) { ... }
```
```